### PR TITLE
[MM-14238] Add make command to generate unsigned x86_64 build for VM/simulator

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Add make command to generate unsigned x86_64 build for VM/simulator.  The resulting app (in zip file) is verified as working for Rainforest.  Generated and submitted zip file can be found here - [Mattermost-unsigned-x86_64.app.zip](https://s3.amazonaws.com/x86-64/ios-1.16.1/Mattermost-unsigned-x86_64.app.zip).

This PR does not include yet the following:
- code signing
- auto-upload to S3

Will address separately if those are necessary to be incorporated in our build workflow.

Note that our existing Android release app is confirmed to be working on Rainforest's VM/emulator.

#### Ticket Link
Jira ticket: [MM-14238](https://mattermost.atlassian.net/browse/MM-14238)

#### Device Information
This PR was tested on: [iOS simulator]

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 assets/override
 dist
+build-ios
```
Comment (enahum) [inline .gitignore]: why is this needed here? once the app builds this folder should be removed entirely, and it seems to do just that in the Make command

File: Makefile [hunk 2]
```diff
@@ -222,6 +222,20 @@ unsigned-ios: stop pre-build check-style ## Build an unsigned version of the iOS
 	@rm -rf build-ios/
 	@ps -ef | grep -i "cli.js start" | grep -iv grep | awk '{print $$2}' | xargs kill -9
 
+unsigned-ios-x86_64: stop pre-build check-style ## Build an unsigned x86_64 version of the iOS app for iPhone simulator
```
Comment (enahum) [inline Makefile]: wondering if the target should be renamed, something like `ios-sim-x86_64` also add the target to the .PHONY at the top of the Makefile

File: Makefile [hunk 3]
```diff
@@ -222,6 +222,20 @@ unsigned-ios: stop pre-build check-style ## Build an unsigned version of the iOS
...
+	@if [ $(shell ps -ef | grep -i "cli.js start" | grep -civ grep) -eq 0 ]; then \
+		echo Starting React Native packager server; \
+		npm start & echo; \
+	fi
+	@echo "Building unsigned x86_64 iOS app for iPhone simulator"
+	@cd fastlane && NODE_ENV=production bundle exec fastlane ios unsigned
+	@mkdir -p build-ios
+	@cd ios/ && xcodebuild -workspace Mattermost.xcworkspace/ -scheme Mattermost -arch x86_64 -sdk iphonesimulator -configuration Release -parallelizeTargets -resultBundlePath ../build-ios/result -derivedDataPath ../build-ios/ ENABLE_BITCODE=NO CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO ENABLE_BITCODE=NO
+	@cd build-ios/Build/Products/Release-iphonesimulator/ && zip -r Mattermost-unsigned-x86_64.app.zip Mattermost.app/
+	@mv build-ios/Build/Products/Release-iphonesimulator/Mattermost-unsigned-x86_64.app.zip .
```
Comment (enahum) [inline Makefile]: should the filename be something like `Mattermost-simulator-x86_64` ?

**Comment by saturninoabril** [comment]
@enahum PR updated per comment.  Also, I added common functions such as `start_packager` and `stop_packager`.

**Comment by saturninoabril** [comment]
I intentionally left them after seeing behaviour of not continuing operation. I suspect it's related to else condition to those targets, but I can take a look further on why.

**Comment by saturninoabril** [comment]
@enahum Added those functions to start and stop targets.  I ran several run, start and build and seems fine. PTAL
